### PR TITLE
Alternc bind9

### PR DESCRIPTION
**Pull request by ulvida**
To allow interaction of the role with the outside code in playbooks and roles, I introduced: 
- the possibility to customize the `named.conf.xxx`, with a variable for the folder where the role takes the template, 
- the register of the `named.conf.xxx` templating task, in order to know if these files were changed or not. 

My needs are for [this project](https://framagit.org/ulvida/belvil.net), where I use this role as well as [ansible-AlternC](https://github.com/UdelaRInterior/ansible-AlternC) which also manages debian bind package (see [this issue](https://framagit.org/ulvida/belvil.net/issues/3) and [this one](https://github.com/UdelaRInterior/ansible-role-bind9/issues/1)).  

Would you accept such PR, considering it's very light?

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -45,6 +45,7 @@
     - named.conf.options
   notify:
     - reload bind9
+  register: named_conf
```
Comment (t2d) [inline tasks/main.yml]: Why do you register this variable?

File: defaults/main.yml [hunk 2]
```diff
@@ -78,6 +78,11 @@ bind9_packages:
     - dnsutils
     - haveged
 
+# Directory for bind9 files templates
+bind9_template_named_conf: bind
+## you can set your own templates, for example with
```
Comment (t2d) [inline defaults/main.yml]: I think this won't pass linting. Please make normal comments. Start with a capital letter.

**Review by t2d — CHANGES_REQUESTED** [review]
If you override `bind` path with a variable, I think you should do it everywhere, don't you?
Please squash your commits in this branch.

File: defaults/main.yml (hunk 2 above)
Comment (ulvida) [reply defaults/main.yml]: Ok, thanks for the tip.

File: tasks/main.yml (hunk 1 above)
Comment (ulvida) [reply tasks/main.yml]: As explained in the PR text, I need to check it outside the role for the interaction between two roles which act on bind9, [here](https://framagit.org/ulvida/belvil.net/blob/master/roles/alternc_bind9/tasks/main.yml#L4) in [this set of playbooks](https://framagit.org/ulvida/belvil.net/). Maybe we can put a comment.

**Comment by ulvida** [comment]
> If you override `bind` path with a variable, I think you should do it everywhere, don't you?

In fact I overrride `templates` path. I re-introduced `bind` and put a `bind9_template` variable for the whole templates tree (by the way I added a `.j2` extension that was missing)
  
> Please squash your commits in this branch.

You're right. done. (sorry for my mess in merging)

File: tasks/main.yml (hunk 1 above)
Comment (t2d) [reply tasks/main.yml]: Alright, please add the comment, than we're good to go.

Comment (t2d) [reply tasks/main.yml]: I think you should rename the variable to something in the bind9 namespace. Then use your playbook to assign the variable `named_conf = bind9_....`

Comment (ulvida) [reply tasks/main.yml]: > I think you should rename the variable to something in the bind9 namespace.

you're right. done, with a comment. 

> Then use your playbook to assign the variable `named_conf = bind9_....`
I can also change my conde outside the role. 

BTW: maybe it's good to manage also a bind9 namespace for handlers names: they can be called outside the role, and ansible doesn't seem to manage any kind of precedence between handlers with the same name.

**Comment by t2d** [comment]
Thank you for your constructive work on this.

**Comment by ulvida** [comment]
> Thank you for your constructive work on this.

thank you to you  for your constructive review.   Union makes force!

**Pull request merged**